app/helpers/social_accounts_helper.rb

```diff
@@ -0,0 +1,28 @@
+module SocialAccountsHelper
+  # rubocop:disable Metrics/CyclomaticComplexity
+  def generate_link(social_account)
+    network = social_account.social_network
+    return "https://github.com/#{social_account.user_name}" if network == 'GitHub'
+    return "https://t.me/#{social_account.user_name}" if network == 'Telegram'
+    return "https://facebook.com/#{social_account.user_name}" if network == 'Facebook'
+    return "https://twitter.com/#{social_account.user_name}" if network == 'Twitter'
+    return "https://gitlab.hpi.de.com/#{social_account.user_name}" if network == 'GitLab@HPI'
+    return 'https://slack.com/' if network == 'Slack'
+    return "https://discordapp.com/#{social_account.user_name}" if network == 'Discord'
+
+    '#'
+  end
+
+  # rubocop:enable Metrics/CyclomaticComplexity
+  def supported_social_networks
+    [
+      %w[GitHub GitHub],
+      %w[Telegram Telegram],
+      %w[Facebook Facebook],
+      %w[Twitter Twitter],
+      ['GitLab@HPI', 'GitLab@HPI'],
+      %w[Slack Slack],
+      %w[Discord Discord]
+    ]
+  end
```

Comment:
What is the sense of this structure - which looks quite redundant at first glance? :-)

See my comment above - I think a single hash object should suffice for everything.

Reply:
That's the way to provide Items to a html select, one string is the string that is displayed and one string is the string that is sent to the backend (normally this would be an id or something similar. I'm not sure if there is a better way. Maybe one could write something like

```ruby
%w[GitHub Telegram Facebook Twitter GitLab@HPI Slack Discord].map do |network|
  [network, network]
end
```

This outputs the following in my irb:

```ruby
[["GitHub", "GitHub"], ["Telegram", "Telegram"], ["Facebook", "Facebook"], ["Twitter", "Twitter"], ["GitLab@HPI", "GitLab@HPI"], ["Slack", "Slack"], ["Discord", "Discord"]]
```

Reply:
Thanks for the explanation. Are the word arrays (`%w`) important for the frontend? Otherwise, your code looks equivalent, so we could eliminate a piece of duplication.

Reply:
`%w` ist just a ruby shorthand for converting a space separated list of words in an array of strings

Reply:
Great, so the `map` proposals works